### PR TITLE
chore: database resource

### PR DESCRIPTION
**Pull request by ecmadao**

### CODE REVIEW

File: provider/resource_database.go [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+	}
+
+	if _, err := c.UpdateDatabase(ctx, patch); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return dataSourceDatabaseRead(ctx, d, m)
+}
+
+func resourceDatabaseDelete(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
```
Comment (d-bytebase) [inline provider/resource_database.go:154]:
```suggestion
func resourceDatabaseDelete(_ context.Context, d *schema.ResourceData, —— interface{}) diag.Diagnostics {
```

**Review by d-bytebase — APPROVED** [review]

File: provider/resource_database.go [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+	}
+
+	if _, err := c.UpdateDatabase(ctx, patch); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return dataSourceDatabaseRead(ctx, d, m)
+}
+
+func resourceDatabaseDelete(_ context.Context, d *schema.ResourceData, —— interface{}) diag.Diagnostics {
```
Comment (d-bytebase) [inline provider/resource_database.go:154]:
```suggestion
func resourceDatabaseDelete(_ context.Context, d *schema.ResourceData, _ interface{}) diag.Diagnostics {
```

File: docs/resources/database.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+The database resource. You can read or update a single database through `bytebase_database` resource.
+
+> For now we don't support create or delete the database, only support change it's project or labels. The databases're auto-sync once the instance is created.
+
+## Example Usage
+
+```terraform
+# Update a database
+resource "bytebase_database" "employee" {
```
Comment (d-bytebase) [inline docs/resources/database.md:19]: Do we have a way to represent the project -> database in the following way?

```
project {
...
  "instances/ins1/databases/db1",
  "instances/ins1/databases/db2",
  "instances/ins2/databases/db1",
  "instances/ins2/databases/db2",
...
}
```

Comment (ecmadao) [reply docs/resources/database.md:19]: Sure, we can optimize the project data source and add the `databases` field

Comment (d-bytebase) [reply docs/resources/database.md:19]: Thanks! That will be easier to manage databases under a project.

**Pull request merged**